### ISSUE TITLE
Change GPU check

### ISSUE THREAD

**Issue by Kaixhin** [NervanaSystems/neon#19]
I've been making Docker images for deep learning libraries, with CPU-only and CUDA-enabled versions. I've successfully made the [base version](https://registry.hub.docker.com/u/kaixhin/neon/) of neon , but my [CUDA version](https://registry.hub.docker.com/u/kaixhin/cuda-neon/) doesn't get built properly. The issue in the Makefile is that `nvidia-smi` is used to check that a CUDA-enabled GPU is available, and whilst this isn't available in Docker build environments it has been possible to build libraries with the installed SDK. If a GPU isn't needed for the build, is it possible to change this test to check for something that is e.g. `nvcc --version`?

**Comment by scttl** [comment]
Yes this seems like a reasonable alternative check.  Thanks for the commit and producing Docker images!

**Comment by Kaixhin** [comment]
Just following up since I've found time to run some models. The CUDA version fails on Kepler GPUs (as expected) but works beautifully on Maxwell GPUs!

**Comment by scttl** [comment]
Great to hear!

As a suggestion you may want to rename the docker image to make it clear that it is Maxwell only since you're using the nervanagpu backend. 

In fact it would probably good to have another docker image based off our cudanet GPU backend since this will work on Kepler cards too.

**Comment by Kaixhin** [comment]
Good point - I've now replaced the CUDA version with [nervanagpu-neon](https://registry.hub.docker.com/u/kaixhin/nervanagpu-neon/) and [cudanet-neon](https://registry.hub.docker.com/u/kaixhin/cudanet-neon/).

As far as I can tell it isn't necessary to install [cuda-convnet2](https://github.com/NervanaSystems/cuda-convnet2) beforehand as the neon installer does it. One weird issue I've had on doing local builds is that when attempting to do a multithreaded make on the cudanet repo it works fine, but it consistently fails when run on the neon repo with the cudanet flag. I'm not going to raise an issue but if you wish to investigate further the most pertinent log I saw was

`read jobs pipe: Resource temporarily unavailable.  Stop`

which looks like somewhere along the line pip isn't handling memory well.
